app/controllers/pokemon_controller.rb

```diff
@@ -15,97 +15,55 @@ class PokemonController < ApplicationController
 
   # GET /pokemon
   def index
-    @pokemon_all = Pokemon.all
     @pokemon = Pokemon.paginate(page: params[:page], per_page: 20)
     
     if @pokemon.empty?
       # puts "here empty"
-      render json: {}, status: :ok
+      render json: {}
     else
-        @pokeList = Hash.new
-      # puts "por el ELSE me fui"
-        @pokemon.each do |u|
-            @pokeList[u.name] = {
-                name: u.name,
-                id: u.id
-            }
-        end
-        @res = {
-            # count_pokes: @pokeList.length
-            all_count_pokes: @pokemon_all.length,
-            total_pages: @pokemon.total_pages,
-            current_page: @pokemon.current_page
-        }
-        if @pokemon.next_page
-          next_page_url = pokemon_url(page: @pokemon.next_page)
-          @res[:next_page] = next_page_url
-        end
-        @res.merge!(@pokeList)
-        render json: @res, status: :ok
+        render json: PokemonIndexService.res(@pokemon, Pokemon.count)
     end
   end
 
   # GET /pokemon/{id}
   def show
-    @poke = find_pokemon
-    @poke_json = @poke.slice(:id, :name, :type_1, :type_1, :type_2, :total, :hp, :attack,
-                :defense, :sp_atk, :sp_def, :speed, :generation, :legendary)
-            render json: @poke_json, status: :ok
+    @poke = PokemonFindService.find(Pokemon, params[:id])
+    if !@poke.nil?
+      @poke_json = @poke.slice(:id, :name, :type_1, :type_1, :type_2, :total, :hp, :attack,
```

Comment:
Could've used the serializer also.

Reply:
Used -- Implemented with different attributes
Merci !!